src/main/java/org/folio/circulation/domain/PatronGroupRepository.java

```diff
@@ -95,6 +93,72 @@ private User addGroupToUser(User user, Map<String, PatronGroup> groupMap) {
       groupMap.getOrDefault(user.getPatronGroupId(), null));
   }
 
+  public CompletableFuture<Result<Loan>> findGroupForLoan(Result<Loan> loanResult) {
+    return loanResult.after(loan ->
```

Comment:
There is a `combineAfter` method for this kind of use case. This code can be replaced with:
```
    return loanResult.combineAfter(loan ->
      getPatronGroupById(loan.getPatronGroupIdAtCheckout()), Loan::withPatronGroup);
```

Reply:
Yes, sure